mixer/adapter/opa/README.md

```diff
@@ -0,0 +1,109 @@
+# Opa adapter
+
+The OPA [mixer adapter](https://istio.io/docs/concepts/policy-and-control/mixer.html#adapters) is an implementation of [authorization template](https://github.com/istio/istio/tree/master/mixer/template/authorization)
+that evaluates the client request using the [Open Policy Agent](http://www.openpolicyagent.org/) engine.
+
+Opa adapter embedded the [Open Policy Agent](http://www.openpolicyagent.org/) as a library inside a Mixer adapter.
+
+![mixer adapter opa](mixer_adapter_opa.png?raw=true)
+
+The adapter is responsible for (1) instantiating an [Open Policy Agent](http://www.openpolicyagent.org/) instance,
+(2) passing the parameters to [Open Policy Agent](http://www.openpolicyagent.org/) and getting the evaluation results from OPA at runtime
+
+## Configuration flow (1 and 2 in the above figure):
+
+  1. Service producer sets authorization rules via istioctl. The rules are saved in Istio Configuration server.
```

Comment:
Service producer sets Open Policy Agent policies (written in REGO) as the adapter parameter.

Reply:
Done